listen.go

```diff
@@ -74,41 +79,19 @@ func (l *listener) SetAddrFilters(fs *filter.Filters) {
 	l.filters = fs
 }
 
-type connErr struct {
-	conn transport.Conn
+type connOrErr struct {
+	conn iconn.Conn
 	err  error
 }
 
 // Accept waits for and returns the next connection to the listener.
-// Note that unfortunately this
-func (l *listener) Accept() (transport.Conn, error) {
-	for con := range l.incoming {
-		if con.err != nil {
-			return nil, con.err
-		}
-
-		c, err := newSingleConn(l.ctx, l.local, "", con.conn)
-		if err != nil {
-			con.conn.Close()
-			if l.catcher.IsTemporary(err) {
-				continue
-			}
-			return nil, err
-		}
-
-		if l.privk == nil || !iconn.EncryptConnections {
-			log.Warning("listener %s listening INSECURELY!", l)
-			return c, nil
-		}
-		sc, err := newSecureConn(l.ctx, l.privk, c)
-		if err != nil {
-			con.conn.Close()
-			log.Infof("ignoring conn we failed to secure: %s %s", err, c)
-			continue
-		}
-		return sc, nil
+func (l *listener) Accept() (iconn.Conn, error) {
+	if l.privk == nil || !iconn.EncryptConnections {
+		log.Warningf("listener %s listening INSECURELY!", l)
 	}
-	return nil, fmt.Errorf("listener is closed")
+
+	c := <-l.incoming
```

Comment:
Actually, I was wrong. We'll also need to check if the channel is closed (`c, ok := <-l.incoming`...).

Reply:
Right. Fixed.